app/scripts/modules/core/cluster/filter/clusterFilter.component.ts

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+import {CLUSTER_FILTER_SERVICE} from 'core/cluster/filter/clusterFilter.service';
+
+import {compact, uniq, map} from 'lodash';
+import {IScope, module} from 'angular';
+import {Application} from '../../application/application.model';
+
+export const CLUSTER_FILTER = 'cluster';
+
+let ngmodule = module(CLUSTER_FILTER, [
+  require('./collapsibleFilterSection.directive.js'),
+  CLUSTER_FILTER_SERVICE,
+  require('./clusterFilter.model.js'),
+  require('../../filterModel/dependentFilter/dependentFilter.service.js'),
+  require('./clusterDependentFilterHelper.service.js'),
+]);
+
+class ClusterFilterCtrl {
+  public categoryHeadings: any;
+  public stackHeadings: string[];
+  public instanceTypeHeadings: any;
+  public regionHeadings: any;
+  public accountHeadings: any;
+  public providerTypeHeadings: any;
+  public availabilityZoneHeadings: any;
+  public app: Application;
+  public tags: any[];
+
+  constructor(public $scope: IScope,
+              public clusterFilterService: any,
+              public ClusterFilterModel: any,
+              public $rootScope: IScope,
+              public clusterDependentFilterHelper: any,
+              public dependentFilterService: any,
+  ) {
+  }
+
+  public $onInit() {
+    const { $scope, $rootScope, ClusterFilterModel, clusterFilterService, app } = this;
+
+    $scope.sortFilter = ClusterFilterModel.sortFilter;
+    this.tags = ClusterFilterModel.tags;
+
+    if (app.serverGroups.loaded) {
+      this.initialize();
+    }
+
+    app.serverGroups.onRefresh($scope, () => this.initialize());
+
+    $scope.$on('$destroy', $rootScope.$on('$locationChangeSuccess', () => {
+      ClusterFilterModel.activate();
+      clusterFilterService.updateClusterGroups(app);
+    }));
+  }
+
+  public updateClusterGroups() {
+    const { dependentFilterService, ClusterFilterModel, clusterDependentFilterHelper, clusterFilterService, app } = this;
+
+    let {providerType, instanceType, account, availabilityZone, region} = dependentFilterService.digestDependentFilters({
+      sortFilter: ClusterFilterModel.sortFilter,
+      dependencyOrder: ['providerType', 'account', 'region', 'availabilityZone', 'instanceType'],
+      pool: clusterDependentFilterHelper.poolBuilder(app.serverGroups.data)
+    });
+
+    this.providerTypeHeadings = providerType;
+    this.accountHeadings = account;
+    this.availabilityZoneHeadings = availabilityZone;
+    this.regionHeadings = region;
+    this.instanceTypeHeadings = instanceType;
+
+    ClusterFilterModel.applyParamsToUrl();
+    clusterFilterService.updateClusterGroups(app);
+  };
+
+  private getHeadingsForOption(option: string) {
```

Comment:
Ah, is that why `call-signature` doesn't flag? It would flag if it can't infer type, but does not if type is easily inferred? If that's the case, I'm okay not forcing it then for simple inferred types.